MAX31855 thermocouple driver

CiprianRegus opened the pull request:
The MAX31855 performs cold-junction compensation
and digitizes the signal from a K-, J-, N-, T-, S-, R-, or
E-type thermocouple. The data is output in a signed
14-bit, SPI-compatible, read-only format. This converter
resolves temperatures to 0.25°C, allows readings as
high as +1800°C and as low as -270°C, and exhibits
thermocouple accuracy of ±2°C for temperatures ranging
from -200°C to +700°C for K-type thermocouples.

Signed-off-by: Ciprian Regus <ciprian.regus@analog.com>

rbolboac commented on drivers/temperature/max31855/iio_max31855.c:
```diff
@@ -0,0 +1,344 @@
+		if (max31855_iio->active_channels & NO_OS_BIT(0)) {
+			i_temp = NO_OS_GENMASK(15, 0) & raw_buff;
+			i_temp = no_os_sign_extend32(i_temp, 15);
+			buff[i] = i_temp;
```
so my assumption is that scan_index 0 corresponds to b0 of active_channels
scan_index 1 corresponds to  b1 of active_channels

is it possible that in this case i_temp is shown when t_temp is selected and vice-versa?

amiclaus commented on drivers/temperature/max31855/iio_max31855.c:
```diff
@@ -0,0 +1,344 @@
+
+	ret = max31855_init(&descriptor->max31855_desc, init_param->max31855_dev_init);
+	if (ret)
+		return ret;
```
memory leak -> `descriptor`

amiclaus commented on drivers/temperature/max31855/iio_max31855.c:
```diff
@@ -0,0 +1,344 @@
+	iio_max31855 = dev;
+	iio_max31855->active_channels = mask;
+
+	if (mask & NO_OS_BIT(0))
```
`hweight` functions from util.c might fit here. (https://github.com/analogdevicesinc/no-OS/blob/78b11d7d6e95fdfc1a985d9ebb7120ce84e15b6e/util/no_os_util.c#L177-L206)

amiclaus commented on drivers/temperature/max31855/max31855.c:
```diff
@@ -0,0 +1,161 @@
+
+	ret = no_os_spi_init(&descriptor->comm_desc, &init_param->spi_init);
+	if (ret)
+		return ret;
```
memory leak: `descriptor`

amiclaus reviewed (commented):
some comments on my side:

CiprianRegus commented:
- Added missing doxygen comments
- Added IIO_MOD_TEMP_AMBIENT modifier name.
- Set the correct active channel indexes is `max31855_iio_read_samples`.
- Fixed memory leaks.
- Used `hweight` instead for active channel count.

adrimbarean reviewed (approved):
Very good and needed changes. I only posted a few comments in the name of pedantry, but it can be merged as is, in my opinion.

CiprianRegus commented:
- Removed `ret` init to 0.
- Removed the masking from `no_of_active_channels`. 
- Fixed pointer placement
